infrastructure/lambda-get-document-nrl.tf

```diff
@@ -0,0 +1,43 @@
+resource "aws_api_gateway_resource" "get_document_reference" {
+  rest_api_id = aws_api_gateway_rest_api.ndr_doc_store_api.id
+  parent_id   = module.create-doc-ref-gateway.gateway_resource_id
+  path_part   = "{id}"
+}
+
+resource "aws_api_gateway_method" "get_document_reference" {
+  rest_api_id      = aws_api_gateway_rest_api.ndr_doc_store_api.id
+  resource_id      = aws_api_gateway_resource.get_document_reference.id
+  http_method      = "GET"
+  authorization    = "NONE"
+  api_key_required = true
```

Comment:
Given we're using a variable `api_key_required` in `infrastructure/modules/gateway/main.tf` that defines it as false, I'd suggest either:

A) We keep it consistent and use either hardcoded true/false values
B) Rename `api_key_required` to something like `api_key_not_required` and create a new variable called `api_key_is_required` for the true value

Reply:
Discussed with Noga & Abbas - happy to move with this as-is as this is a one-off exception to a general pattern used all over.